File: ClientApp/src/app/interfaces/security/userForCreationDto.ts
import { UserRolesDto } from "./user-roles-dto";

export interface UserForCreationDto {
	userName: string;
	email: string;
	password: string;
	confirmPassword: string;
	birthDate: string;
	rolesId: UserRolesDto[];
	promotionalMails: boolean;
}

File: Contracts/Model/Security/UserForCreationDto.cs
using System.ComponentModel.DataAnnotations;

namespace Contracts.Model.Security;

public class UserForCreationDto
{
	public string UserName { get; set; }
	public string Email { get; set; }
	public string Password { get; set; }
	[Compare(nameof(Password))]
	public string ConfirmPassword { get; set; }
	public DateOnly BirthDate { get; set; }
	public IEnumerable<UserRoleDto> RolesId { get; set; }
	public bool PromotionalMails { get; set; }
}